optaplanner-core/src/main/java/org/optaplanner/core/api/score/Score.java

```diff
@@ -155,6 +165,22 @@
      */
     boolean isCompatibleArithmeticArgument(Score otherScore);
 
+    /**
+     * A {@link PlanningSolution} is feasible if it has no broken hard constraints
+     * and {@link #isSolutionInitialized()} is true.
+     *
+     * Simple scores ({@link SimpleScore}, {@link SimpleLongScore}, {@link SimpleBigDecimalScore}) are always feasible.
+     *
+     * @return true if the hard score is 0 or higher and the {@link #getInitScore()} is 0.
+     */
+    default boolean isFeasible() {
+        /*
+         * This exception will only be thrown for custom scores that did not implement FeasibilityScore.
+         * TODO Safe to remove this default implementation once FeasibilityScore has been removed in 8.0.
+         */
+        throw new UnsupportedOperationException();
+    }
```

Comment:
Noteworthy, I anticipate comments here.

Reply:
I am fine with this approach. When I added toShortString() in FEB 2017 I should have probably done it this way too with a default method until 8.0 is.

Please add this in the list of PLANNER-750 TODO's, to remove make this method abstract again.